tests/mgr/test_restful_operations_pool_endpoint.py

```diff
@@ -0,0 +1,138 @@
+import json
+
+from api.exceptions import ConfigError, OperationFailedError
+from api.restful.pool import Pool
+from cli.cephadm.cephadm import CephAdm
+from utility.log import Log
+
+LOG = Log(__name__)
+
+
+# Request header
+HEADER = {
+    "Accept": "application/vnd.ceph.api.v1.0+json",
+    "Content-Type": "application/json",
+}
+
+# Missing config parameter err msg
+CONFIG_ERR_MSG = "Mandatory config parameter '{}' not provided"
+
+
+def _get(config, osd_node, url, endpoint, username, key, status_code):
+    # Check if any <args> are present
+    if config.get("args"):
+        # Get pool name from config
+        pool_name = config.get("args")
+        # Get ID for the given pool name
+        out = json.loads(CephAdm(osd_node).ceph.osd.pool.ls(format="json"))
+        if not out:
+            raise OperationFailedError("Failed to fetch the pools")
+        pool_id = None
+        for pool in out:
+            if pool["pool_name"] == pool_name:
+                pool_id = pool["pool_id"]
+                break
+        if not pool_id:
+            raise OperationFailedError(f"Failed to get pool_id for {pool_name}")
+        endpoint += f"/{pool_id}"
+    _code, _ = Pool(url=url, api=endpoint).get(
+        username=username, key=key, check_sc=True
+    )
+    if _code == status_code:
+        LOG.info("RESTful API validation is successful")
+        return 0
+    raise OperationFailedError(
+        f"RESTful GET method for {endpoint} failed with status code:{_code}"
+    )
+
+
+def _post(config, url, endpoint, username, key, status_code):
+    data = config.get("data")
+    if not data:
+        raise ConfigError(CONFIG_ERR_MSG.format("data"))
+    # Use POST method on the /pool endpoint of the RESTful module
+    _code, _ = Pool(url=url, api=endpoint).post(
+        username=username, key=key, data=data, check_sc=True
+    )
+    if _code == status_code:
+        LOG.info("RESTful API validation is successful")
```

Comment:
this statement is not required, kindly remove.

Reply:
Done, removed.